datalad/core/distributed/push.py

```diff
@@ -572,19 +489,28 @@ def _push(dspath, content, target, data, force, jobs, res_kwargs, pbars,
             )
             return
 
-    # make sure that we always push the active branch (the context for the
-    # potential path arguments) and the annex branch -- because we claim
-    # to know better than any git config
-    must_have_branches = [active_branch] if active_branch else []
-    if is_annex_repo:
-        must_have_branches.append('git-annex')
-    for branch in must_have_branches:
-        _append_branch_to_refspec_if_needed(ds, refspecs2push, branch)
+    #
+    # We know where to push to, honor dependencies
+    # XXX we could do this right after we know the value of `target`,
+    # but this owuld mean we would also push to dependencies
+    # even when no actual push to the primary target is needed
```

Comment:
> but this would mean we would also push to dependencies even when no actual push to the primary target is needed

But wouldn't that be desired? If I configure a publication dependency, isn't the aim to have both targets "in sync"? The dependency could require a push to make it so even if the primary target doesn't. May be because I only just added the dependency and now want to make sure the secondary target catches up?

To be sure, just confirmed, that this would currently work as I described.

Reply:
I am not sure what the question is. Yes, it is desired. But it has nothing to do with this PR. The comment is about making more obvious that the structure of the function could be further simplified, but a potential behavior change is implied -- but again, nothing that this PR (against maint!) wants to change.